notifications-service/src/main/java/mk/ukim/finki/notifications_service/service/job/MailScheduler.java

```diff
@@ -0,0 +1,47 @@
+package mk.ukim.finki.notifications_service.service.job;
+
+import mk.ukim.finki.notifications_service.model.PostNotification;
+import mk.ukim.finki.notifications_service.model.enumeration.MailStatus;
+import mk.ukim.finki.notifications_service.service.MailService;
+import mk.ukim.finki.notifications_service.service.PostNotificationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@EnableScheduling
+@Component
+public class MailScheduler {
+
+    @Autowired
+    private MailService mailService;
+
+    @Autowired
+    private PostNotificationService notificationService;
+
+    @Scheduled(cron = "0 */1 * ? * *")
```

Comment:
Team, eventually the service should be configured to run on multiple instances, we shouldn't have internal cron scheduling because every instance of the service will run the independent schedule right?! And if the service logic is not dealing with this, we'll end up with multiple notifications when we need one... So we maybe need an external cron service, as we had for labs?! https://github.com/stefankochev/soa-labs/pull/4, check the diagram please.